addons/mail/static/src/discuss/gif_picker/common/gif_picker.js

```diff
@@ -166,27 +166,32 @@ export class GifPicker extends Component {
             return;
         }
         this.state.loadingGif = true;
-        try {
-            const params = {
-                country: this.userService.lang.slice(3, 5),
-                locale: this.userService.lang,
-                search_term: this.state.searchTerm,
-            };
-            if (this.next) {
-                params.position = this.next;
-            }
-            const { results, next } = await this.rpc("/discuss/gif/search", params, {
+        const params = {
+            country: this.userService.lang.slice(3, 5),
+            locale: this.userService.lang,
+            search_term: this.state.searchTerm,
+        };
+        if (this.next) {
+            params.position = this.next;
+        }
+        const res = await this.sequential(() =>
+            this.rpc("/discuss/gif/search", params, {
                 silent: true,
-            });
-            if (results) {
-                this.next = next;
-                for (const gif of results) {
-                    this.pushGif(gif);
-                }
+            }).catch(() => {
+                this.state.loadingError = true;
+            })
+        );
+        if (!res) {
+            return;
+        }
+        const { next, results } = res;
+        if (results && next) {
+            this.next = next;
+            for (const gif of results) {
+                this.pushGif(gif);
             }
-        } catch {
-            this.state.loadingError = true;
         }
+        this.state.loadingError = false;
```

Comment:
seems a little bit weird to handle err like this. Could change to something like
```js
        this.sequential(() => this.rpc("/discuss/gif/search", params, {silent: true}))
            .then(({ next, results } = {}) => {
            if (results && next) {
                this.next = next;
                for (const gif of results) {
                    this.pushGif(gif);
                }
            }
        }).catch(() => {
            this.state.loadingError = true;
        });
            this.state.loadingGif = false;
        ```

Reply:
This isn't equivalent.
I what to catch error on the RPC, but not on the sequential. This is will think there is an error because sequential can resolve unwanted RPC. This is why the `catch` is on `this.rpc` and not `this.sequential`.

Reply:
I do not fully understand what is `This is will think there is an error because sequential can resolve unwanted RPC. ` If you mean `sequential` can still keep going if there it's an err already being thrown. I think maybe it's a better design to let `sequential` throw the err rather than resolve and continue going.

Reply:
I think you are right, I changed the PR :)